client/thirdparty.go

```diff
@@ -0,0 +1,54 @@
+package client
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	k8sApiUnversioned "k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/restclient"
+)
+
+// ThirdPartyClient can be used to access third party resources
+type ThirdPartyClient struct {
+	*restclient.RESTClient
+	baseURL string
+}
+
+func (c *ThirdPartyClient) Workflows(namespace string) WorkflowInterface {
+	return newWorkflows(c, namespace)
+}
+
+// NewThirdparty creates a new ThirdPartyClient
+func NewThirdparty(gv *k8sApiUnversioned.GroupVersion, c *restclient.Config) (*ThirdPartyClient, error) {
+	config := *c
+	groupVersion := *gv
+	if err := setThirdPartyDefaults(&groupVersion, &config); err != nil {
+		return nil, err
+	}
+	client, err := restclient.RESTClientFor(&config)
+	if err != nil {
+		return nil, err
+	}
+	baseURL := c.Host + config.APIPath + "/" + config.GroupVersion.Group + "/" + config.GroupVersion.Version
+	return &ThirdPartyClient{client, baseURL}, nil
+}
+
+// Configuration for RESTClient
+func setThirdPartyDefaults(groupVersion *k8sApiUnversioned.GroupVersion, config *restclient.Config) error {
+	config.APIPath = "/apis"
+	if config.UserAgent == "" {
+		config.UserAgent = restclient.DefaultKubernetesUserAgent()
+	}
+
+	config.GroupVersion = groupVersion
+
+	//config.Codec = thirdpartyresourcedata.NewCodec(client.NewExtensions(config).RESTClient.Codec(), gvk.Kind)
```

Comment:
Die syntax kwam ik tegen in de kubectl code. Ik denk dat je dat nodig hebt als je de ingebakken List wilt gebruiken, maar met alleen dat ben je er niet. Daarom gewoon werkende code aangehouden.